crates/language/src/lib.rs

```diff
@@ -97,6 +97,56 @@ macro_rules! impl_lang_expando {
   };
 }
 
+/// Implements the `ALIAS` associated constant for the given lang, which is
+/// then used to define the `alias` const fn and a `Deserialize` impl.
+/// Also generates as convenience conversions between the lang types
+/// and `SupportedType`.
+macro_rules! impl_aliases {
+  ($($lang:ident => $as:expr),* $(,)?) => {
+    $(
+      impl $lang {
+        pub const ALIAS: &'static [&'static str] = $as;
+      }
+
+      impl fmt::Display for $lang {
+        fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+          write!(f, "{:?}", self)
+        }
+      }
+
+      impl<'de> Deserialize<'de> for $lang {
+        fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+        where
+          D: Deserializer<'de>,
+        {
+          let s = String::deserialize(deserializer)?;
```

Comment:
`String::deserialize` will cause allocations here. Is it possible to deserialize it to &str?

Reply:
Probably, I had used the `SupportLang`'s impl as a starting point, which is how that ended up there.

Reply:
Addressed in 2444388. I chose the more verbose `Visitor` pattern for this so it'd work with transient strings, which have a more relaxed lifetime requirement. `<&str>::deserialize` would be another option, but it will cause an error with certain `Deserializer`s due to the stricter lifetime requirement.